docs/en/stack/ml/nlp/ml-nlp-text-emb-vector-search-example.asciidoc

```diff
@@ -0,0 +1,339 @@
+
+[discrete]
+[[ex-te-vs-requirements]]
+== Requirements
+
+include::ml-nlp-shared.asciidoc[tag=nlp-requirements]
+
+
+[discrete]
+[[ex-te-vs-deploy]]
+== Deploy a text embedding model
+
+include::ml-nlp-shared.asciidoc[tag=nlp-eland-clone-docker-build]
+
+Select a text embedding model from the 
+{ml-docs}/ml-nlp-model-ref.html#ml-nlp-model-ref-ner[third-party model reference list]. 
+This example uses the 
+https://huggingface.co/sentence-transformers/msmarco-MiniLM-L-12-v3[msmarco-MiniLM-L-12-v3]
+sentence-transformer model.
+
+Install the model by running the `eland_import_model_hub` command in the Docker 
+image:
+
+[source,shell]
+--------------------------------------------------
+docker run -it --rm elastic/eland \
+    eland_import_hub_model \
+      --cloud-id $CLOUD_ID \
+      -u <username> -p <password> \
+      --hub-model-id sentence-transformers/msmarco-MiniLM-L-12-v3 \
+      --task-type text_embedding \
+      --start
+--------------------------------------------------
+
+You need to provide an administrator username and password and replace the 
+`$CLOUD_ID` with the ID of your Cloud deployment. This Cloud ID can be copied 
+from the deployments page on your Cloud website.
+
+include::ml-nlp-shared.asciidoc[tag=nlp-start]
+
+include::ml-nlp-shared.asciidoc[tag=nlp-sync]
+
+[discrete]
+[[ex-text-emb-test]]
+== Test the text embedding model
+
+Deployed models can be evaluated in {kib} under **{ml-app}** > 
+**Trained Models** by selecting the **Test model** action for the respective 
+model.
+
+[role="screenshot"]
+image::images/ml-nlp-text-emb-test.png[Test trained model UI]
+
+.**Test the model by using the _infer API**
+[%collapsible]
+====
+You can also evaluate your models by using the 
+{ref}/infer-trained-model-deployment.html[_infer API]. In the following request, 
+`text_field` is the field name where the model expects to find the input, as 
+defined in the model configuration. By default, if the model was uploaded via 
+Eland, the input field is `text_field`.
+
+[source,js]
+--------------------------------------------------
+POST /_ml/trained_models/sentence-transformers__msmarco-minilm-l-12-v3/_infer
+{
+  "docs": {
+    "text_field": "How is the weather in Jamaica?"
+  }
+}
+--------------------------------------------------
+
+The API returns a response similar to the following:
+
+[source,js]
+--------------------------------------------------
+{
+  "inference_results": [
+    {
+      "predicted_value": [
+        0.39521875977516174,
+        -0.3263707458972931,
+        0.26809820532798767,
+        0.30127981305122375,
+        0.502890408039093,
+        ...
+      ]
+    }
+  ]
+}
+--------------------------------------------------
+// NOTCONSOLE
+====
+
+The result is the predicted dense vector transformed from the example text.
+
+
+[discrete]
+[[ex-text-emb-data]]
+== Load data
+
+The data set msmarco-passagetest2019-top1000.tsv (you can find it 
```

Comment:
> The data set msmarco-passagetest2019-top1000.tsv (you can find it 

It seems like there are links to two different tsv files in this paragraph, which makes me a little uncertain which one I'm supposed to use. If we're *not* expecting users to download this first tsv file, could we link to an info page or somehow make it clearer that we're linking to this only for information purposes?